src/lib.rs

```diff
@@ -47,115 +45,342 @@ pub enum Protection {
+    /// for a mutable version.
+    ///
+    /// If the protection has not been [set explicitly](#method.protection), this method
+    /// assumes [`Read`](enum.Protection.html#variant.Read).
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    pub fn map(&self) -> Result<Mmap> {
+        let mut this = *self;
+        if this.protection.is_none() {
+            this.protection = Some(Protection::Read);
+        }
+        let inner = this.map_inner()?;
+        Ok( Mmap { inner } )
+    }
+
+    /// Actually map this anonymous mapping into the address space.
+    ///
+    /// This method returns a mutable mapping, see [`map()`](#method.map) for an immutable
+    /// version.
+    ///
+    /// If the protection has not been [set explicitly](#method.protection), this method
+    /// assumes [`ReadWrite`](enum.Protection.html#variant.ReadWrite).
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    ///
+    /// This method *also* returns `Err` with `ErrorKind` set to `InvalidInput` if the specified
+    /// protection does not allow the mapping to be mutable.
+    pub fn map_mut(&self) -> Result<MmapMut> {
+        let mut this = *self;
+        if this.protection.is_none() {
+            this.protection = Some(Protection::ReadWrite);
+        }
+        match this.protection.unwrap() {
+            Protection::Read | Protection::ReadExecute => Err(Error::new(
+                ErrorKind::InvalidInput,
+                "Invalid protection for a mutable mapping",
+            )),
+            Protection::ReadWrite | Protection::ReadCopy => {
+                let inner = this.map_inner()?;
+                Ok( MmapMut { inner } )
+            }
+        }
+    }
+}
+
+// File-backed mappings
+
+/// Options that can be used to configure how a file-backed mapping is created.
+///
+/// Create this structure by calling [`memmap::file()`](fn.file.html),
+/// then chain call methods to configure additional options, finally, call [`map()`](#method.map)
+/// or [`map_mut()`](#method.map_mut).
+#[derive(Copy, Clone, Debug)]
+pub struct FileMappingOptions<'a> {
+    file: &'a File,
+    protection: Option<Protection>,
+    offset: usize,
+    len: Option<usize>,
+}
 
-        options
+/// Configure a new file-backed mapping.
+///
+/// # Unsafety
+///
+/// On Unix, this function is `unsafe`, because it's up to the caller to ensure
+/// that no other process or thread is accessing the same file concurrently.
+/// In particular, it is **undefined behavior** in Rust for the memory to be
+/// modified by some other code while there's a reference to it.
+///
+/// Note, however, that using this with
```

Comment:
This has never really been clear to me.  The Linux `mmap` manpage says:

> It is unspecified whether changes made to the file after the mmap() call are visible in the mapped region.

That doesn't seem like it will play well with Rust's safety semantics.